services/merak-network/activities/node_register_activity.go

```diff
@@ -0,0 +1,56 @@
+/*
+MIT License
+Copyright(c) 2022 Futurewei Cloud
+    Permission is hereby granted,
+    free of charge, to any person obtaining a copy of this software and associated documentation files(the "Software"), to deal in the Software without restriction,
+    including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and / or sell copies of the Software, and to permit persons
+    to whom the Software is furnished to do so, subject to the following conditions:
+    The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+    WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+package activities
+
+import (
+	"context"
+	"encoding/json"
+	pb "github.com/futurewei-cloud/merak/api/proto/v1/merak"
+	"github.com/futurewei-cloud/merak/services/merak-network/entities"
+	"github.com/futurewei-cloud/merak/services/merak-network/http"
+	"github.com/futurewei-cloud/merak/services/merak-network/utils"
+	"log"
+	"sync"
+)
+
+func RegisterNode(ctx context.Context, compute []*pb.InternalComputeInfo, wg *sync.WaitGroup, projectId string) (string, error) {
+	log.Println("RegisterNode")
+	//defer wg.Done()
+	log.Printf("compute %s", compute)
+	nodeInfo := entities.NodeStruct{}
+
+	for _, host := range compute {
+		log.Printf("host %s", host)
+		nodeBody := entities.NodeBody{
+			LocalIP:    host.Ip,
+			MacAddress: host.Mac,
+			NodeID:     host.Id,
+			NodeName:   host.Name,
+			ServerPort: 50001,
+			Veth:       host.Veth,
+		}
+		nodeInfo.Hosts = append(nodeInfo.Hosts, nodeBody)
+	}
+	log.Printf("nodeInfo: %s", nodeInfo)
+	returnMessage, returnErr := http.RequestCall("http://"+utils.ALCORURL+":30007/nodes/bulk", "POST", nodeInfo, nil)
+	if returnErr != nil {
+		log.Printf("returnErr %s", returnErr)
+	}
```

Comment:
@yanmo96  you should return error `if returnErr != nil`, otherwise, `returnMessage` will be nil and your next code will be failed.